### ISSUE TITLE
Thank u！

### ISSUE THREAD

**Issue by AKAxiaobai-123** [rafiibnsultan/GeoSAM#2]
Hello author, I read your article with great interest, as I am also a researcher in this field. I would like to know how the h5 file was generated. Could you please share some insights on this? Thank you for your assistance.

**Comment by rafiibnsultan** [comment]
Hello,

Thank you for showing interest.

The process to create an h5 file is straightforward. Begin by selecting a domain-specific encoder from a segmentation network that you intend to use as an auxiliary model. Ensure this encoder is applied to your chosen dataset. Rather than preserving the output images, the procedure involves storing the feature embeddings within an h5 file, typically the outputs from the layer just before the last layer. For an illustrative example, you might refer to the Tile2Net folder; we've modified the original source code to capture the embeddings, however, the approach is applicable to any domain-specific encoder.

**Comment by AKAxiaobai-123** [comment]
Thank you for your prompt response.
I went down to study the source code you modified in tilenet, and I only saw that you modified the ocrnet of the network, is that correct? However, I noticed that this encoder is not used anywhere. How exactly do you use it? Do you take out the encoder separately to load the weights and perform forward inference? 

I'm really looking forward to your reply, as this is very important for my experiment.

**Comment by rafiibnsultan** [comment]
Sorry for the late reply.
You are right, when we run Tile2Net on our training dataset, we save the image embeddings of each of the images and save it into the disc (what you see the process in ocrnet). Later, we use the embeddings.

**Comment by AKAxiaobai-123** [comment]
Hello, I am very pleased to see your response. 
I have carefully studied it using my own road dataset. However, I found that during the inference process, my Intersection over Union (IoU) value is quite low. After a thorough examination of your code, I discovered that the mask values are generated through Tile2Net. Could you please explain the specific process of how they are generated? 
I am eagerly looking forward to your reply.Thanks！

**Comment by rafiibnsultan** [comment]
Hi,
You can check the Tile2Net’s official repository. With some tweaking of their code you can save their semantic segmentation results (I believe we also have uploaded the tweaked code from our side in this repository).
One other thing, you can use any other pre trained network. For example, you can train a UNet in your own dataset to do this same thing as well. Because prompts which are just a guide to SAM, don't have to be fully accurate.

Hope this helps!

**Comment by AKAxiaobai-123** [comment]
Okay, thank you for your answer, I will give it a try, and I will let you know if I get good results.

**Comment by AKAxiaobai-123** [comment]
I have trained according to your code, but the results obtained are not satisfactory, and I do not know the reason for this. The intersection over union (IoU) I obtained with the mask_decoder I trained is only 0.14, whereas with the decoder you trained, it is 0.41. Below are the result images obtained with your weights. Could you please take a look and see if there are any issues?
Additionally, I have another question for you: Are the images in the masks folder that you trained with obtained through automated segmentation by SAM ?
Looking forward to your reply, thank you!
![bj100001_sat_input](https://github.com/rafiibnsultan/GeoSAM/assets/84052575/5f1a1598-7dc2-43a7-bf69-c28cd1a127e6)
…]()
![bj100001_mask](https://github.com/rafiibnsultan/GeoSAM/assets/84052575/7ff8bfe6-17f3-464c-9073-9f2eb6e96389)
![bj100001_sat](https://github.com/rafiibnsultan/GeoSAM/assets/84052575/67cfe709-9b28-4a1f-8420-3a2d02168f2b)



![output_image_0](https://github.com/rafiibnsultan/GeoSAM/assets/84052575/ab199747-e3ac-40d1-89f3-27a50e000468)

**Comment by rafiibnsultan** [comment]
Hi,
Sorry for the late reply. Let me reopen the issue so I don't miss it next. You might not get the best result with the decoder we trained if it doesn't match the classes that you are working on. You might further fine-tune it using your own dataset.

I didn't get the second question, what masks?

**Comment by AKAxiaobai-123** [comment]
Thanks！
I am not sure how the data in your dataset's masks file is described or how this mask is generated.
Thank you once again for your answer. I am also curious to know if having multiple classes, such as seven, would result in less effective training outcomes?
![image](https://github.com/rafiibnsultan/GeoSAM/assets/84052575/2fe61e60-df41-4d30-b7a4-f5adf9242313)

**Comment by rafiibnsultan** [comment]
We have uploaded a demo. check here:  [demo.ipynb](https://github.com/rafiibnsultan/GeoSAM/blob/GeoSAM_with_text/demo/example.ipynb) Also, the demo folder in the "GeoSAM_with_text" has some examples of the dataset. 

We also had multiple classes, 3 to be exact, if you want to have 7 you just have to change the one-hot-encoding according to it.

**Comment by rafiibnsultan** [comment]
Hi,

Since you work in a related domain, I was wondering if you would be interested in reviewing this current work as a paper that we plan to submit to a relevant journal. If you are open to this, I can include you as a potential reviewer. For this, I would need your contact information or a link to your Google Scholar profile.

If you're willing to be a potential reviewer, please feel free to reach out to me via email at: hm4013@wayne.edu.
